keps/sig-scheduling/2891-simplified-config/README.md

```diff
@@ -0,0 +1,622 @@
+
+Therefore, a way for users to simply toggle all of a plugin's extension points
+(or a subset of the extension points as defined necessary by the plugin developer)
+would remove complexity and the potential for misconfiguration.
+
+### Goals
+
+* Provide an option for simplified plugin config from the end-user's perspective
+* Enable this config in all in-tree default scheduler plugins
+
+### Non-Goals
+
+* Refactor or introduce any breaking changes to the scheduling framework
+
+## Proposal
+
+A new extension point, `MultiPoint` (final name TBD) will be made available to
+users and developers which handles plugin registration for multiple extension
+points simultaneously.
+
+From the user's perspective, this will be a new (optional) field in the scheduler's
+config to enable or disable plugins across multiple extension points.
+
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a cluster administrator, I wish to enable plugin `XYZ` for all applicable extension points (ie, `Filter` 
+and `Score`). I do not care about internal implementation or extension points such as 
+`PreFilter`, nor do I want to have to update my config should plugin `XYZ` change its 
+implementation within the scheduling framework to use different extension points.
+
+```yaml
+apiVersion: kubescheduler.config.k8s.io/v1beta3
+kind: KubeSchedulerConfiguration
+profiles:
+  - schedulerName: default-scheduler
+    plugins:
+      multiPoint:
+        enabled:
+        - name: "XYZ"
+          weight: 3
+```
+
+#### Story 2
+
+As a cluster administrator, I wish to enable plugin `XYZ` for only *some* of its applicable extension 
+points (ie `Filter` but *not* `Score`). I still do not care about extension points such 
+as `PreFilter`. I want to have control over which subsets of extension points are enabled.
+
+```yaml
+apiVersion: kubescheduler.config.k8s.io/v1beta3
+kind: KubeSchedulerConfiguration
+profiles:
+  - schedulerName: default-scheduler
+    plugins:
+      multiPoint:
+        enabled:
+        - name: "XYZ"
+      preScore:
+        disabled:
+        - "*"
+      score:
+        disabled:
+        - "*"
+```
+
+#### Story 3
+
+As a plugin developer, I want users to be able to easily enable my plugin for all applicable 
+extension points (or a subset of extension points as defined either by me or my users' needs). 
+I want this available with minimal code updates required from me.
+
+### Notes/Constraints/Caveats (Optional)
+
+* changes should have minimal effect on existing plugins and users
+
+### Risks and Mitigations
+
+* out-of-tree plugins may be developed by anyone, so the information that is 
+passed to them and received from them should be kept to a minimum. However these are 
+unsupported so security is ultimately up to users to be aware of.
+* UX will need to be carefully examined, as a new config field can introduce confusing 
+and unexpected outcomes if different combinations of options are not well-defined.
+
+## Design Details
+
```

Comment:
Also part of the Design is what happens with the defaults. Today, when users export the component config (via `write-config-to`), they get a full detail of each of the extension points. Will the defaults be migrated to the `MultiPoint` extension point?

Reply:
This depends on at what point in the code `write-config-to` operates. If it's before framework initialization, then users would get the `MultiPoint` extension exported. If it's after the `MultiPoint` extension is unrolled, then they will get the old config.

Reply:
So you are saying that the default configuration now will use `MultiPoint` instead of individual extension points.

We should write the config after unrolling.

Reply:
Yes, that's the case. As long as we update the default config to use MultiPoint (I had an example of this), then that's what will get printed

Reply:
Sounds good. Please add this as part of the Design Details.